src/orchestrator/orchestrator.rs

```diff
@@ -265,12 +267,26 @@ impl Borrow<ArtifactPath> for ProducedArtifact {
 
 impl<'a> Orchestrator<'a> {
     pub async fn run(self, output: &mut Vec<ArtifactPath>) -> Result<HashMap<Uuid, Error>> {
-        let (results, errors) = self.run_tree().await?;
+        let token = CancellationToken::new();
+        let cloned_token = token.clone();
+
+        tokio::spawn(async move {
+            info!("Received the ctl-c signal, stopping...");
```

Comment:
Looking at the code I would assume that this always gets printed - shouldn't you move this below `tokio::signal::ctrl_c().await`?

Reply:
Yeah, you're right. I will move it